src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1028 @@
+          size="sm"
+        />
+        {question?.questionOptions?.rendering === 'date' && (
+          <Toggle
+            id={`toggle-allow-future-date-${ruleId}`}
+            labelText="Allow Future dates"
+            hideLabel
+            toggled={isAllowFutureDate}
+            onToggle={handleAllowFutureDateChange}
+            size="sm"
+          />
+        )}
+        {question?.questionOptions?.rendering === 'number' && (
+          <Toggle
+            id={`toggle-disallow-decimal-value-${ruleId}`}
+            labelText="Disallow Decimal Value"
+            hideLabel
+            toggled={isDisallowDecimals}
+            onToggle={handleDisallowDecimalValueChange}
+            size="sm"
+          />
+        )}
+      </div>
+    );
+  },
+);
+
+interface RuleConditionProps {
+  fieldId: string;
+  questions: Array<Question>;
+  answers: Array<Record<string, string>>;
+  isTablet: boolean;
+  launchDeleteConditionalLogicModal: (deleteAll: boolean) => void;
+  deleteCondition: () => void;
+  addNewConditionalLogic: () => void;
+  isNewCondition: boolean;
+  isNewRule: boolean;
+  addCondition: () => void;
+  index: number;
+  handleConditionChange: (id: string, field: string, value: string, index: number) => void;
+  conditions: Array<Condition>;
+}
+
+export const RuleCondition = React.memo(
+  ({
+    fieldId,
+    isTablet,
+    questions,
+    isNewCondition,
+    isNewRule,
+    index,
+    deleteCondition,
+    addCondition,
+    launchDeleteConditionalLogicModal,
+    answers,
+    handleConditionChange,
+    conditions,
+    addNewConditionalLogic,
+  }: RuleConditionProps) => {
+    const { t } = useTranslation();
+    const answer = answers.filter((answer) => answer !== undefined);
+    const [isConditionValueVisible, setIsConditionValueVisible] = useState<boolean>(
+      Boolean(conditions[index]?.targetValue),
+    );
+    const handleSelectCondition = (selectedCondition: string) => {
+      setIsConditionValueVisible(!['Is Empty', 'Not Empty'].includes(selectedCondition));
+    };
+    const [inputValue, setInputValue] = useState(conditions[index]?.targetValue || '');
+    const handleValueChange = (selectedItem: string) => {
+      handleConditionChange(fieldId, 'targetValue', selectedItem, index);
+      setInputValue(selectedItem);
+    };
+
+    return (
+      <div className={styles.ruleSetContainer}>
+        <div className={styles.sectionContainer}>
+          {isNewCondition ? (
+            <Dropdown
+              id={`logicalOperator-${index}`}
+              className={styles.logicalOperator}
+              initialSelectedItem={conditions[index]?.logicalOperator || 'and'}
+              defaultSelectedItem="and"
+              items={['and', 'or']}
+              onChange={({ selectedItem }: { selectedItem: string }) => {
+                handleConditionChange(fieldId, `logicalOperator`, selectedItem, index);
+              }}
+              size={isTablet ? 'lg' : 'sm'}
```

Comment:
We could abstract this logic out into a variable and reuse it:

```
const responsiveSize = isTablet ? 'lg' : 'sm';
```

And then:

```
size={responsiveSize}
```

So we have a single source of truth for easier refactors.